### DISCUSSION TITLE
Are the results of twig filters on translations cached?

### DISCUSSION THREAD

**Discussion by hellomedia** [symfony/symfony#52504] [Q&A]
```
{{ 'foo'|trans|capitalize }}
```

Is the capitalization done on every request, or is this process cached?

If I am trying to reduce my impact on the planet (energy consumption) as a developer, should I consider embedding the capitalization inside the translation itself to avoid unnecessary processing?

**Comment by 94noni** [comment]
You can try to find the cached twig php file generated from your original twig template and see how it is generated 

but regarding the initial goal of code computation reduction, to me it is irrelevant perhaps go check other part of application (generally reducing i/o, network calls, etc etc)
